core/adapters/random.go

```diff
@@ -56,33 +69,76 @@ func (ra *Random) TaskType() models.TaskType {
 
 // Perform returns the the proof for the VRF output given seed, or an error.
 func (ra *Random) Perform(input models.RunInput, store *store.Store) models.RunOutput {
-	key, err := getKey(ra, input)
+	key, preSeed, blockHash, blockNum, err := getInputs(ra, input, store)
 	if err != nil {
-		return models.NewRunOutputError(errors.Wrapf(err, "bad key for vrf task"))
-	}
-	seed, err := getSeed(input)
-	if err != nil {
-		return models.NewRunOutputError(errors.Wrap(err, "bad seed for vrf task"))
+		return models.NewRunOutputError(err)
 	}
-	solidityProof, err := store.VRFKeyStore.GenerateProof(key, seed)
+	solidityProof, err := store.VRFKeyStore.GenerateProof(key, preSeed,
+		*blockHash, *blockNum)
 	if err != nil {
 		return models.NewRunOutputError(err)
 	}
 	ethereumByteArray := fmt.Sprintf("0x%x", utils.EVMEncodeBytes(solidityProof[:]))
 	return models.NewRunOutputCompleteWithResult(ethereumByteArray)
 }
 
-// getSeed returns the numeric seed for the vrf task, or an error
-func getSeed(input models.RunInput) (*big.Int, error) {
+// getInputs parses the JSON input for the values needed by the random adapter,
+// or returns an error.
+func getInputs(ra *Random, input models.RunInput, store *store.Store) (
+	key *vrfkey.PublicKey, preSeed *vrf.Seed, blockHash *common.Hash,
+	blockNum *uint64, err error) {
+	key, err = getKey(ra, input)
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrapf(err, "bad key for vrf task")
```

Comment:
I wonder if this might do better to return some sort of struct that holds the input fields instead of 4 nils?

Reply:
Thanks, I introduced a `vrf.SeedData` struct to pass around these values.